packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js

```diff
@@ -63,6 +75,45 @@ const ProductFullDetail = props => {
         />
     ) : null;
 
+    // Fill a map with field/section -> error.
+    const errors = new Map();
+    if (derivedErrorMessage) {
+        Object.keys(ERROR_MESSAGE_TO_FIELD_MAPPING).forEach(key => {
+            if (derivedErrorMessage.includes(key)) {
+                const target = ERROR_MESSAGE_TO_FIELD_MAPPING[key];
+                const message = ERROR_FIELD_TO_MESSAGE_MAPPING[target];
+                errors.set(target, message);
+            }
+        });
+        if (!errors.size) {
+            errors.set(
+                'form',
+                'Could not add item to cart. Please check required options and try again.'
+            );
+        }
+    }
+
+    // If there are form errors, display and scroll to them.
+    const errorsRef = useRef(null);
+    const formErrors = errors.get('form');
+    let formErrorSection;
```

Comment:
Minor but we've started naming JSX sections that may or may not be populated with the prefix `maybe`.

This one would be `maybeFormErrorSection`.

Reply:
8dcd4f9